tests/toranj/cli/test-019-network-data-publisher-compact-routes.py

```diff
@@ -0,0 +1,304 @@
+    verify(len(routes) == 3)
+    verify('2600:1:2:3::/64 s med ' + r1_rloc in routes)
+    verify('fc00::/7 s high ' + r1_rloc in routes)
+    verify('fc00::/7 s high ' + r2_rloc in routes)
+
+
+verify_within(check_netdata_step6, 2)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# Change the preference of last route on `r1` from `high` to `low`.
+# Check that `r1` continues to use the compact prefix but switches
+# back to `med` preference.
+
+r1.netdata_publish_route('fd01:4::/64', 's', 'low')
+
+
+def check_netdata_step7():
+    routes = r2.get_netdata_routes()
+    verify(len(routes) == 3)
+    verify('2600:1:2:3::/64 s med ' + r1_rloc in routes)
+    verify('fc00::/7 s med ' + r1_rloc in routes)
+    verify('fc00::/7 s high ' + r2_rloc in routes)
+
+
+verify_within(check_netdata_step7, 2)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# Unpublish `2600:1:2:3::/64` route on `r1`. Make sure it does not
+# impact the compact prefix.
+
+r1.netdata_unpublish('2600:1:2:3::/64')
+
+
+def check_netdata_step8():
+    routes = r2.get_netdata_routes()
+    verify(len(routes) == 2)
+    verify('fc00::/7 s med ' + r1_rloc in routes)
+    verify('fc00::/7 s high ' + r2_rloc in routes)
+
+
+verify_within(check_netdata_step8, 2)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# Unpublish two prefixes on `r1`. This causes number of routes to go
+# below the threshold. Check that all original prefixes are added
+# back.
+
+r1.netdata_unpublish('fd01:1::/64')
+r1.netdata_unpublish('fd01:2::/64')
+
+
+def check_netdata_step9():
+    routes = r2.get_netdata_routes()
+    verify(len(routes) == 3)
+    verify('fd01:3:0:0::/64 s med ' + r1_rloc in routes)
+    verify('fd01:4:0:0::/64 s low ' + r1_rloc in routes)
+    verify('fc00::/7 s high ' + r2_rloc in routes)
+
+
+verify_within(check_netdata_step9, 2)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# Publish a NAT64 prefix on `r1. Make sure it is not counted
+# towards optimizable routes.
+
+r1.netdata_publish_route('fd01:5::/96', 'sn')
+
+
+def check_netdata_step10():
+    routes = r2.get_netdata_routes()
+    verify(len(routes) == 4)
+    verify('fd01:3:0:0::/64 s med ' + r1_rloc in routes)
+    verify('fd01:4:0:0::/64 s low ' + r1_rloc in routes)
+    verify('fd01:5:0:0:0:0::/96 sn med ' + r1_rloc in routes)
+    verify('fc00::/7 s high ' + r2_rloc in routes)
+
+
+verify_within(check_netdata_step10, 2)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# Publish `fc00::7` itself on `r1`.
+
+r1.netdata_publish_route('fc00::/7', 's')
+
+
+def check_netdata_step11():
+    routes = r2.get_netdata_routes()
+    verify(len(routes) == 3)
```

Comment:
Where did the routes `fd01:3:0:0::/64` and `fd01:4:0:0::/64` go to?  Does this mean that more-specific routes are automatically suppressed, in case of ULA routes?  Why do this for ULAs but not for non-ULA prefixes then? Seems like we should discuss this in SPEC-1130.

Reply:
Maybe it counts N=3 ULA prefixes now, based on the newly added fc00::/7 ? And then all prefixes are optimized into fc00::/7.